exporter/clickhousetracesexporter/clickhouse_exporter.go

```diff
@@ -300,9 +301,14 @@ func newStructuredSpan(otelSpan ptrace.Span, ServiceName string, resource pcommo
 			IsColumn: false,
 		}
 		if v.Type() == pcommon.ValueTypeDouble {
-			numberTagMap[k] = v.Double()
-			spanAttribute.NumberValue = v.Double()
-			spanAttribute.DataType = "float64"
+			if !math.IsNaN(v.Double()) {
+				numberTagMap[k] = v.Double()
+				spanAttribute.NumberValue = v.Double()
+				spanAttribute.DataType = "float64"
+			} else {
+				zap.S().Warn("NaN value in tag map skipping: ", zap.String("key", k), zap.Float64("value", v.Double()))
```

Comment:
nit, the value is already NaN

```suggestion
				zap.S().Warn("NaN value in tag map, skipping key", zap.String("key", k))
```